### ISSUE TITLE
Delay with Dolby Digital

### ISSUE THREAD

**Issue by m-koptev** [JarnoVgr/SonosTricks#1]
Thanks for the guide! I found that after all manipulations, there occurs a significant delay of 500ms if the dolby digital is active. I tried reducing the sound quality for the virtual sound recorder, no result. Do you experience the same? Is there any workaround?

**Comment by GingerAdonis** [comment]
For me personally the delay varied a lot per open game or app.

I've switched to hacked Realtek audio drivers in order to enable Dolby Digital encoding from optical out on any motherboard. Then I use the optical out to HDMI converter, which is included with the Sonos Beam, to be able to connect quite a long HDMI cable (about 20m.) to the Sonos Beam directly (and circumvent the TV for audio). 
The downsides of this method are:
- No HDMI-CEC when using the HDMI audio cable from the PC.
- Manually switching HDMI cables whenever I want to game or watch TV.

**However! The benefit of using this method is that there's literally no delay in audio as Realtek chips have hardware encoding.**

If you'd like to give this method a shot as well I can write up a new guide. Just let me know. :)

**Comment by nilsbyte** [comment]
@GingerAdonis I would like to know how it works with Realtek drivers. I have a Asrock X300 Mini board (HTPC) with a Realtek Onboard sound solution.

**Comment by GingerAdonis** [comment]
Here's a link to the hacked Realtek drivers that can enable Dolby Digital 5.1 on optical output for any motherboard. https://github.com/AlanFinotty/AAFDCHDriverMod/releases

Make sure to remove the original Realtek drivers beforehand. :)

**Comment by nilsbyte** [comment]
I use the ASRock X300 Mini which only has HDMI output and no dedicated sound or graphics card. I am using the iGPU from the AMD Ryzen CPU. 

This means I have to use HDMI for video and audio output. I don't understand the inner workings of HDMI audio on PC. Which part handles/should handle Dolby Digital encoding? It needs to be Dolby Digital as it's the only format Sonos accepts.

**Comment by GingerAdonis** [comment]
> This means I have to use HDMI for video and audio output. I don't understand the inner workings of HDMI audio on PC. Which part handles/should handle Dolby Digital encoding? It needs to be Dolby Digital as it's the only format Sonos accepts.
The following guide should also work for your situation: https://github.com/GingerAdonis/SonosTricks/blob/master/surroundSoundOnWindows/README.md

**Comment by nilsbyte** [comment]
Yes @GingerAdonis but in this solution I have audio delay of some hundred milliseconds. I guess it’s because it’s a software encoder?

**Comment by GingerAdonis** [comment]
> Yes @GingerAdonis but in this solution I have audio delay of some hundred milliseconds. I guess it’s because it’s a software encoder?

You're correct. I'm not aware of a work-around, unfortunately. Please note that regular stereo sound doesn't have this delay.
